crates/sdk/src/keygen/mod.rs

```diff
@@ -413,6 +413,8 @@ impl AggProvingKey {
         reader: &impl Halo2ParamsReader,
         pv_handler: &impl StaticVerifierPvHandler,
     ) -> Self {
+        let wall_timer = std::time::Instant::now();
```

Comment:
@nyunyunyunyu note: you should move your `Timer` util into stark-sdk because it seems generally useful. Although I think @shuklaayush knows a way to just let normal `tracing` spans do this kind of thing automatically

Reply:
this is kind of hacky but i was suggesting something like this to create a custom tracing layer to handle the timing metric calculation
https://gist.github.com/shuklaayush/da4efef3cdae3a7031209847f9f0f1c3